File: ParkingPlaceServer/ParkingPlaceServer/DTO/ReservationAndPaidParkingPlacesDTO.cs
using Newtonsoft.Json;
using ParkingPlaceServer.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ParkingPlaceServer.DTO
{
	public class ReservationAndPaidParkingPlacesDTO
	{
		[JsonProperty("reservation")]
		public ReservationDTO Reservation { get; set; }

		[JsonProperty("regularPaidParkingPlace")]
		public PaidParkingPlaceDTO RegularPaidParkingPlace { get; set; }

		[JsonProperty("paidParkingPlacesForFavoritePlaces")]
		public List<PaidParkingPlaceDTO> PaidParkingPlacesForFavoritePlaces { get; set; }
		

		public ReservationAndPaidParkingPlacesDTO()
		{

		}

		public ReservationAndPaidParkingPlacesDTO(Reservation r, PaidParkingPlace rp, List<PaidParkingPlace> ps)
		{
			if (r == null)
			{
				Reservation = null;
			}
			else
			{
				Reservation = new ReservationDTO(r);
			}

			if (rp == null)
			{
				RegularPaidParkingPlace = null;
			}
			else
			{
				RegularPaidParkingPlace = new PaidParkingPlaceDTO(rp);
			}

			if (ps == null || (ps != null && ps.Count == 0))
			{
				PaidParkingPlacesForFavoritePlaces = new List<PaidParkingPlaceDTO>();
			}
			else
			{
				PaidParkingPlacesForFavoritePlaces = ps.Select(p => new PaidParkingPlaceDTO(p)).ToList();
			}
			
		}

	}
}

File: ParkingPlace/app/src/main/java/com/rmj/parking_place/dto/ReservationAndPaidParkingPlacesDTO.java
package com.rmj.parking_place.dto;

import java.util.List;

public class ReservationAndPaidParkingPlacesDTO {
    private ReservationDTO reservation;
    private PaidParkingPlaceDTO regularPaidParkingPlace;
    private List<PaidParkingPlaceDTO> paidParkingPlacesForFavoritePlaces;

    public ReservationAndPaidParkingPlacesDTO() {

    }

    public ReservationDTO getReservation() {
        return reservation;
    }

    public void setReservation(ReservationDTO reservation) {
        this.reservation = reservation;
    }

    public PaidParkingPlaceDTO getRegularPaidParkingPlace() {
        return regularPaidParkingPlace;
    }

    public void setRegularPaidParkingPlace(PaidParkingPlaceDTO regularPaidParkingPlace) {
        this.regularPaidParkingPlace = regularPaidParkingPlace;
    }

    public List<PaidParkingPlaceDTO> getPaidParkingPlacesForFavoritePlaces() {
        return paidParkingPlacesForFavoritePlaces;
    }

    public void setPaidParkingPlacesForFavoritePlaces(List<PaidParkingPlaceDTO> paidParkingPlacesForFavoritePlaces) {
        this.paidParkingPlacesForFavoritePlaces = paidParkingPlacesForFavoritePlaces;
    }
}
